src/util/strencodings.cpp

```diff
@@ -508,17 +509,36 @@ std::string Capitalize(std::string str)
     return str;
 }
 
+namespace {
+
+using ByteAsHex = std::array<char, 2>;
+
+constexpr std::array<ByteAsHex, 256> CreateByteToHexMap()
+{
+    std::array<ByteAsHex, 256> byte_to_hex{};
+    for (size_t i = 0; i < byte_to_hex.size(); ++i) {
+        char const* hex = "0123456789abcdef";
```

Comment:
Any reason for not using the constexpr array as removed below?
```
    static constexpr char hexmap[16] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
```

Reply:
It can't be `static` here because I'm using that function in a `constexpr` way. With `static constexpr char...` I get that error message:

```
util/strencodings.cpp:520:31: error: static variable not permitted in a constexpr function
        static constexpr char hexmap[16] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };
                              ^
```

I can use `constexpr char hexmap[16] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' };` though. I just used a string though because it's less typing :slightly_smiling_face:

Reply:
Huh, TIL

> I can use constexpr char hexmap[16] = { '0', '

I prefer that, then. I mean it's not used as a C string so it shouldn't be a C string. The amount of typing shouldn't be a consideration :stuck_out_tongue: